code/game/gamemodes/vampire/vampire.dm

```diff
@@ -439,6 +439,14 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 		check_sun()
 	if(istype(owner.loc.loc, /area/chapel) && !get_ability(/datum/vampire_passive/full))
 		vamp_burn(7)
+	if(istype(owner.loc.loc, /area/lavaland) && !get_ability(/datum/vampire_passive/full)) 
```

Comment:
So you know, `owner.loc.loc` would theoretically allow you to bypass the burn damage if you hide in something like a locker for example. Since `owner.loc.loc` while in a locker would be the turf the locker is on.

Reply:
also @matttheficus bear in mind that most vampire miners take the IV bags from the medical room and use as many as they can on their first victim in the dorms area, ignoring completely this restriction

Reply:
loc.loc swapped to get_area(owner), fixing locker issue

added burns on t2 blood to make it less fun to do so, while making t3 MUCH scarier (90% burn chance)